internal/connector/authn.go

```diff
@@ -104,6 +104,10 @@ func (j *authenticator) getJWK() (*jose.JSONWebKey, error) {
 	}
 	defer res.Body.Close()
 
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
```

Comment:
3xx codes aren't generally errors, but the Go http client handles redirects. So if we ever get a 3xx code here it means the redirect loop was cancelled, so that is an error.

We do the same thing in our api.Client: https://github.com/infrahq/infra/blob/main/api/client.go#L50